scipy/stats/tests/test_stats.py

```diff
@@ -6454,85 +6456,105 @@ def fun(a, axis, weights):
         check_equal_pmean(a, p, desired, axis=axis, weights=weights, rtol=1e-5)
 
 
+@array_api_compatible
+@pytest.mark.usefixtures("skip_if_array_api")
 class TestGeometricStandardDeviation:
     # must add 1 as `gstd` is only defined for positive values
-    array_1d = np.arange(2 * 3 * 4) + 1
-    gstd_array_1d = 2.294407613602
+    array_1d = np.arange(2 * 3 * 4) + 1.
+    gstd_array_1d = np.float64(2.294407613602)
     array_3d = array_1d.reshape(2, 3, 4)
 
-    def test_1d_array(self):
-        gstd_actual = stats.gstd(self.array_1d)
-        assert_allclose(gstd_actual, self.gstd_array_1d)
+    def test_1d_array(self, xp):
+        gstd_actual = stats.gstd(xp.asarray(self.array_1d))
+        xp_assert_close(gstd_actual, xp.asarray(self.gstd_array_1d))
 
+    @pytest.mark.skip_if_array_api("all", reasons="Array API doesn't accept tuples")
```

Comment:
I don't think that this is correct, it should be `np_only` - we want to maintain array-like input for the NumPy backend, otherwise there would be too much BC-break

Reply:
Maybe I need to form a mental model of the plan going forward - eventually, we will remove the environment variable, but not change any of these tests? So yeah, I suppose we want it to run NumPy only then.

Reply:
Yes, this is something that has changed since Pamphile's original PR - he had a `skip_if_array_api` marker which was just an alias for `pytest.mark.skipif(SCIPY_ARRAY_API)`. Ralf pointed out that we'd rather not have that, as we want almost every test to continue to pass with NumPy, hence I removed it when I reworked the infra and effectively replaced it with `np_only`. The exception to this is types which we are actually dropping, like `numpy.matrix`, masked arrays etc.

I've spent a while forming said mental model 😅 and of course the exact timeline for dropping the environment variable isn't set yet, I suppose it depends on how far behind / ahead we are of the array libraries.

Reply:
OK, so it sounds like the meaning of `pytest.mark.skip_if_array_api` has changed completely, so what do you think of a new name like `pytest.mark.skip_backends`?

Reply:
See my other comment. Perhaps `skip_array_api_backends`, although the line lengths are already quite long...

Reply:
I would be happy with `skip_xp_backends`